ee/tabby-ui/app/(dashboard)/(logs)/jobs/components/job-row.tsx

```diff
@@ -0,0 +1,177 @@
+          activeClass="border-red-500 text-xs text-red-500"
+          tooltip="Failed"
+        />
+      </div>
+    </LoadingWrapper>
+  )
+}
+
+export default function JobRow({ name }: { name: string }) {
+  const RECENT_DISPLAYED_SIZE = 10
+
+  const [{ data, fetching }] = useQuery({
+    query: listJobRuns,
+    variables: {
+      last: RECENT_DISPLAYED_SIZE,
+      jobs: [name]
+    }
+  })
+
+  const edges = data?.jobRuns?.edges
+  const displayJobs = useMemo(() => {
+    return edges?.slice().reverse()
+  }, [edges])
+  const lastJob = displayJobs?.[0]
+  const lastFinishedJob = displayJobs?.find(job => Boolean(job.node.finishedAt))
+  const lastSuccessAt = lastFinishedJob
+    ? moment(lastFinishedJob.node.finishedAt).format('YYYY-MM-DD HH:mm')
+    : null
+  return (
+    <LoadingWrapper
+      loading={fetching}
+      fallback={
+        <TableRow>
+          <TableCell colSpan={4}>
+            <ListRowSkeleton />
+          </TableCell>
+        </TableRow>
+      }
+    >
+      <TableRow className="h-16">
+        <TableCell className="font-bold">{name}</TableCell>
+        <TableCell>
+          <div className="flex">
+            {displayJobs?.map(job => {
+              const { createdAt, finishedAt } = job.node
+              const startAt =
+                createdAt && moment(createdAt).format('MMMM D, YYYY h:mm a')
+              const duration =
+                createdAt &&
+                finishedAt &&
+                humanizerDuration(
+                  moment
+                    .duration(moment(finishedAt).diff(createdAt))
+                    .asMilliseconds()
+                )
+              return (
+                <TooltipProvider delayDuration={0} key={job.node.id}>
+                  <Tooltip>
+                    <TooltipTrigger asChild>
+                      <Link
+                        href={`/jobs/detail?id=${job.node.id}`}
+                        className={cn(
+                          'mr-1 h-6 w-2 rounded-full hover:opacity-70',
+                          {
+                            'bg-blue-500': isNil(job.node.exitCode),
+                            'bg-green-500': job.node.exitCode === 0,
+                            'bg-red-500': job.node.exitCode === 1
+                          }
+                        )}
+                      />
+                    </TooltipTrigger>
+                    <TooltipContent>
+                      {startAt && <p>{startAt}</p>}
+                      {duration && <p>Duration: {duration}</p>}
+                    </TooltipContent>
+                  </Tooltip>
+                </TooltipProvider>
+              )
+            })}
+          </div>
+        </TableCell>
+        <TableCell>
+          <Link
+            href={`/jobs/detail?id=${lastJob?.node.id}`}
+            className="flex items-center hover:underline"
+          >
+            <p>{lastSuccessAt}</p>
+            <IconExternalLink className="ml-1" />
```

Comment:
use better icon (external link usually means "open new tab")

Reply:
I removed icon and added a underline because I couldn't find a suitable icon

![屏幕截图 2024-03-29 083644](https://github.com/TabbyML/tabby/assets/5305874/078a237f-1d99-4126-9cce-4d51b8c41a89)